crates/resolver/src/spawn_bg.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2015-2019 Benjamin Fry <benjaminfry@me.com>
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+use std::future::Future;
+
+use futures::future;
+use tokio::{self, runtime::Handle};
+
+use crate::proto::error::ProtoError;
+
+/// A trait for spawning resolver background tasks
+pub trait SpawnBg: Clone + Send + Sync + Unpin + 'static {
```

Comment:
@chunyingw I'd love your input on this type. I'd like to make the JoinHandle fully exposed, but it still needs some work...

Reply:
In the previous code there were embedded spawns in the ConnectionProvider. Were you able to abstract purely around that and not have to worry about the spawns that were happening there? If that’s better for you, I can keep ConnectionProvider as the only abstraction point.

 For some reason it never occurred to me that would have been an easier way for you to generically define the connections in the system. But it makes sense that might be better in general.

Since this is primarily for use cases like yours, if that’s your preference, we can do that, I just want the spawn to be more obvious than it was before.

Reply:
Yes, I was able to purely abstract for TCP and UDP. would you point me which "spawns" you meant, plz?

Reply:
Here, it happens in ConnectionConnect: https://github.com/bluejekyll/trust-dns/blob/cd7ffa58858ac2d842a5a869ebf13703611b73b3/crates/resolver/src/name_server/connection_provider.rs#L155

Reply:
That's good point!! I neglected as tokio executor can be compiled on Fuchsia, but it should be replaced by Fuchsia's executor.

Reply:
@chunyingw see the current changes I put up last night. I think ConnectionProvider is the correct abstraction. Let me know what you think. It's simpler than doing both that and the SpawnBg...

I think you can still use the generic interfaces from proto that you created before with that. See the TokioConnectionProvider for an example. You could create a FuchsiaConnectionProvider for your use cases.